fuel-vm/src/interpreter/alu/wideint.rs

```diff
@@ -0,0 +1,365 @@
+            fn [<from_prim_ $t:lower>](value: primitive_types::[<$t:upper>]) -> $t {
+                let mut buffer = [0u8; ::core::mem::size_of::<$t>()];
+                value.to_little_endian(&mut buffer);
+                $t::from_le_bytes(buffer)
+            }
+
+            /// Drops higher half of the value and converts to `u128` or `ethnum::U256`
+            fn [<truncate_from_prim_ $t:lower>](value: primitive_types::[<$wider_t:upper>]) -> $t {
+                const S: usize = ::core::mem::size_of::<$t>();
+                let mut buffer = [0u8; 2 * S];
+                value.to_little_endian(&mut buffer);
+                let truncated: [u8; S] = buffer[..S].try_into().unwrap_or_else(|_| unreachable!());
+                $t::from_le_bytes(truncated)
+            }
+
+            impl<S, Tx> Interpreter<S, Tx>
+            where
+                Tx: ExecutableTransaction,
+            {
+                pub(crate) fn [<alu_wideint_cmp_ $t:lower>](
+                    &mut self,
+                    ra: RegisterId,
+                    b: Word,
+                    c: Word,
+                    args: CompareArgs,
+                ) -> Result<(), RuntimeError> {
+                    let (SystemRegisters { pc, .. }, mut w) = split_registers(&mut self.registers);
+                    let dest: &mut Word = &mut w[ra.try_into()?];
+
+                    // LHS argument is always indirect, load it
+                    let lhs: $t = $t::from_be_bytes(read_bytes(&self.memory, b)?);
+
+                    // RHS is only indirect if the flag is set
+                    let rhs: $t = if args.indirect_rhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, c)?)
+                    } else {
+                        c.into()
+                    };
+
+                    *dest = [<cmp_ $t:lower>](lhs, rhs, args.mode);
+
+                    inc_pc(pc)
+                }
+
+                pub(crate) fn [<alu_wideint_op_ $t:lower>](
+                    &mut self,
+                    dest_addr: Word,
+                    b: Word,
+                    c: Word,
+                    args: MathArgs,
+                ) -> Result<(), RuntimeError> {
+                    let owner_regs = self.ownership_registers();
+                    let (SystemRegisters { flag, mut of, mut err, pc, .. }, _) = split_registers(&mut self.registers);
+
+                    // LHS argument is always indirect, load it
+                    let lhs: $t = $t::from_be_bytes(read_bytes(&self.memory, b)?);
+
+                    // RHS is only indirect if the flag is set
+                    let rhs: $t = if args.indirect_rhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, c)?)
+                    } else {
+                        c.into()
+                    };
+
+                    let (wrapped, overflow) = [<op_overflowing_ $t:lower>](lhs, rhs, args);
+
+                    if overflow && !is_wrapping(flag.into()) {
```

Comment:
should not panic if `F_UNSAFEMATH` is set

> If the [F_UNSAFEMATH](https://github.com/FuelLabs/fuel-vm/pull/424/index.md#flags) flag is set, an ALU operation that would have panicked will instead set $err to true.

https://github.com/FuelLabs/fuel-specs/blob/813faf0b94bc45cd5e9b55826b57831d27bed394/src/vm/instruction_set.md?plain=1#L144

Reply:
See below